modules/imap/functions.php

```diff
@@ -1495,16 +1495,15 @@ function forward_dropdown($output,$reply_args) {
 if (!hm_exists('parse_sieve_config_host')) {
 function parse_sieve_config_host($host) {
     $url = parse_url($host);
-    $host = $url['host'] ?? $url['path'];
+    if(!isset($url['host'])) {
+        $host = $url['path'];
+    }
     $port = $url['port'] ?? '4190';
-    $scheme = $url['scheme'] ?? 'tcp://';
-    $tls = $scheme === 'tls';
```

Comment:
With migadu we don't add `tls://` at the begininng but with Stalwart if it's implicit we must add it.